### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ncdc**
TODO:
- [x] Link to this from README.md after #554 merges

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    - `rootKcpClient`
+    - `orgKubeClient`
+- Configs intended for `NewClusterForConfig` (i.e. today often called "admin workspace config") should uniformely be called `clusterConfig`
+    - Note: with org workspaces, the second use for the admin workspaces will go away
+    - Note 2: sometimes we use clients for same purpose, but this can be harder to read
+- Cluster-aware clients should follow similar naming conventions:
+    - `crdClusterClient`
+    - `kcpClusterClient`
+    - `kubeClusterClient`
+- `clusterName` is a kcp term. It is **NOT** a name of a physical cluster. If we mean the latter, use `pclusterName` or similar.
```
Comment (marun) [inline CONTRIBUTING.md:77]: (No action required) Why is it important for kcp's `clusterName` to be unqualified? vs using `lclusterName` for a logical cluster and `pclusterName` for physical clusters?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+- In the syncer: upstream = kcp, downstream = pcluster. Depending on direction, "from" and "to" can have different meanings. `source` and `sink` are synonyms for upstream and downstream.
```
Comment (marun) [inline CONTRIBUTING.md:78]: (No action required) source/sink tends to be used to refer to event producers and consumers. Maybe source/destination would be more appropriate for our kcp?

**Review by marun — COMMENTED** [review]
LGTM, thank you for getting the ball rolling!

File: CONTRIBUTING.md (hunk 1 above)
Comment (ncdc) [reply CONTRIBUTING.md:77]: @sttts wrote these, but I'll hazard a guess: the guidance is to never use `clusterName` to mean a physical cluster. If/when unqualified, `clusterName` means a logic cluster.

I maybe want to also suggest that we never use `lcluster...` or `pcluster...` - it's too easy to wonder if that was intended, or if the leading `l`/`p` was a typo.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ncdc) [reply CONTRIBUTING.md:78]: My preference would be target instead of sink. What do y'all think?

File: CONTRIBUTING.md (hunk 1 above)
Comment (sttts) [reply CONTRIBUTING.md:77]: `clusterName` is defined by kube's ObjectMeta. Also kcp is technically about clusters in the lcluster sense. Pcluster is something on-top. We are just not that far yet to split it appart and have the syncer in the same code base.

File: CONTRIBUTING.md (hunk 2 above)
Comment (sttts) [reply CONTRIBUTING.md:78]: target of status syncer is kcp. Don't like that.

Comment (ncdc) [reply CONTRIBUTING.md:78]: Fair

Comment (ncdc) [reply CONTRIBUTING.md:78]: Then I'd vote for src/dest

Comment (sttts) [reply CONTRIBUTING.md:78]: or just kcp and pcluster.

**Review by sttts — APPROVED** [review]

**Pull request merged**